contracts/MerkleLib.sol

```diff
@@ -37,6 +39,41 @@ library MerkleLib {
         return nodes[0];
     }
 
+    function merkleRootWithMinTree(
+        bytes memory data,
+        uint256 chunkSize
+    ) internal pure returns (bytes32) {
+        uint256 n = (data.length + chunkSize - 1) / chunkSize;
+        /* Round up to next 2^n */
+        uint256 nChunks = (n <= 1) ? 1 : BinaryRelated.findNextPowerOf2(n);
```

Comment:
We don't need to find the next power of 2 - just allocate nodes with n, and when then merge two nodes (in line 68-70), if the last node is odd, then merge it with bytes32(0).

Reply:
Okay. But data.length == 0 is still a corner case.

Reply:
Yep.  Just return bytes32(0) when data.length == 0 is a good practice.